src/Mentat.UI/Controllers/HomeController.cs

```diff
@@ -6,21 +6,52 @@
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.Extensions.Logging;
 using Mentat.UI.Models;
+using Mentat.UI.Services;
+using Mentat.UI.ViewModels;
 
 namespace Mentat.UI.Controllers
 {
     public class HomeController : Controller
     {
         private readonly ILogger<HomeController> _logger;
+        private readonly ICardService cardService;
 
-        public HomeController(ILogger<HomeController> logger)
+        public HomeController(ILogger<HomeController> logger, ICardService cardService)
         {
             _logger = logger;
+            this.cardService = cardService;
         }
 
+        static List<Card> GetRandomCards(List<Card> cards, int numCards)
+        {
+            if (cards.Count < numCards)
+            {
+                return cards;
+            }
+
+            HashSet<int> indices = new HashSet<int>();
+            Random random = new Random();
+
+            List<Card> randomCards = new List<Card>();
+            while (indices.Count < numCards)
+            {
+                int index = random.Next(cards.Count);
+                if (!indices.Contains(index))
+                {
+                    indices.Add(index);
+                    randomCards.Add(cards[index]);
+                }
+            }
```

Comment:
Import the namespace System.Linq;

Replace line 32 thru 46 with

```csharp
return cards.OrderBy(a => Guid.NewGuid()).Take(numCards);
```

Reply:
side note, don't use the letter 'a' in my example (I copy pasted that). Use 'card', and if you don't switch the return to IEnumerable<Card> you will need to also add a ToList() after the Take